common-api/src/main/java/org/jboss/galleon/diff/FsEntryFactory.java

```diff
@@ -162,7 +162,7 @@ private void initChildren(final FsEntry parent) throws IOException {
         }
         if(hasDirs) {
             for(FsEntry child : parent.getChildren()) {
-                if(!child.dir) {
+                if(!child.dir || !Files.isReadable(child.p)) {
```

Comment:
@spyrkob , is it enough? DirectoryStream<Path> stream = Files.newDirectoryStream(parent.p) (line 155) should fail if the installation root directory contains non readable directory. No?

Reply:
@jfdenise my understanding is that if the parent directory has read permission, you're able to list all the children. The problem happens when trying to create the `newDirectoryStream` for the child folder without read permission (line 168).